packages/did-provider-ebsi/src/types.ts

```diff
@@ -43,5 +42,26 @@ export interface ICreateIdentifierArgs {
   kms?: string
   alias?: string
   type?: EbsiDidSpecInfo
-  options?: IKeyOpts
+  options?: {
+    methodSpecificId?: string // method specific id for import
+    secp256k1?: IKeyOpts | VerificationMethod
+    secp256r1?: IKeyOpts | VerificationMethod
+  }
+}
+
+export interface ICreateIdentifierOpts {}
+
+export enum KeyType {
```

Comment:
Don't introduce new key type enums. We already have TKeyType. If you need to restrict the current Enum, then use a Pick<>

Reply:
`TKeyType` is an `UnionType` and `Pick<>` doesn't work with it and according what I've read so far, it doesn't work either with `enums`. In addition `enums` are completely different from `UnionTypes` and the process to extract a subset of keys from them is completely different. :confused: In the end I just had to use `Extract<>`.